### PR TITLE
Make `--follow` the default in OSS

### PR DESCRIPTION
**Pull request by elyousfi5**
If neither follow or unfollow is provided the default is to follow all the agents.

https://github.com/Ostorlab/ostorlab/assets/144013278/adc748d7-2a2f-4dd9-b7d9-9827cfb344d6

####
If unfollow is True, we don't follow any agent.

https://github.com/Ostorlab/ostorlab/assets/144013278/a4045251-419f-4aab-bcfb-ea45105ffc91

####
If follow is provided, we follow only the provided agents.

https://github.com/Ostorlab/ostorlab/assets/144013278/a99eb6b2-e797-4174-acab-997e79d645f0

### CODE REVIEW

**Review by oussamaessaji — COMMENTED** [review]
Unit test?

**Review by 3asm — CHANGES_REQUESTED** [review]

File: src/ostorlab/cli/scan/run/run.py [hunk 1]
```diff
@@ -63,8 +63,8 @@
     "--install", "-i", help="Install missing agents.", is_flag=True, required=False
 )
 @click.option(
-    "--follow",
-    help="Follow logs of provided list of agents and services.",
+    "--unfollow",
+    help="Don't show logs of provided list of agents and services.",
```
[the remark this answers is not in the document]
Comment (BlueSquare1) [reply src/ostorlab/cli/scan/run/run.py:67]: if it's boolean that would mean users can't decide to exclude only certain agent logs

[the remark this answers is not in the document]
Comment (elyousfi5) [reply src/ostorlab/cli/scan/run/run.py:67]: @BlueSquare1 No we will keep follow, the logic will be as follow:
 # If unfollow is True, we don't follow any agent.
 # If follow is provided, we follow only the provided agents.
 # If neither follow or unfollow is provided the default is to follow all the agents.

File: src/ostorlab/cli/scan/run/run.py [hunk 2]
```diff
@@ -68,6 +67,13 @@
     multiple=True,
     default=[],
 )
+@click.option(
+    "--no-follow",
```
Comment (benyissa) [inline src/ostorlab/cli/scan/run/run.py:71]:
```suggestion
    "--unfollow",
```
--unfollow is more intuitive for me.

**Comment by elyousfi5** [comment]
> Unit test?

I added some videos as a demonstration.

File: src/ostorlab/cli/scan/run/run.py
Comment (mohsinenar) [inline src/ostorlab/cli/scan/run/run.py:1]: Can you add a unit test?

File: src/ostorlab/cli/scan/run/run.py [hunk 3]
```diff
@@ -130,8 +137,12 @@ def run(
             console.error(f"{e}")
             raise click.ClickException("Invalid asset Group Definition.") from e
     runtime_instance: runtime.Runtime = ctx.obj["runtime"]
-    # set list of log follow.
-    runtime_instance.follow = follow
+
+    # Prepare and set the list of agents to follow.
```
Comment (3asm) [inline src/ostorlab/cli/scan/run/run.py:141]: The code needs unit test.

File: src/ostorlab/cli/scan/run/run.py [hunk 4]
```diff
@@ -168,6 +179,32 @@ def run(
         )
 
 
+def _prepare_agents_to_follow(
+    agent_keys: List[str], follow: List[str], unfollow: bool
```
Comment (3asm) [inline src/ostorlab/cli/scan/run/run.py:183]:
```suggestion
    agent_keys: List[str], follow: List[str], no_follow: bool
```

Keep it consistent.

**Review by 3asm — CHANGES_REQUESTED** [review]

**Pull request merged**